mobile/lib/ui/viewer/file/zoomable_image.dart

```diff
@@ -348,4 +366,36 @@ class _ZoomableImageState extends State<ZoomableImage> {
   }
 
   bool _isGIF() => _photo.displayName.toLowerCase().endsWith(".gif");
+
+  Future<void> _loadInSupportedFormat(File file) async {
+    _logger.info("Compressing ${_photo.displayName} to viewable format");
+    _convertToSupportedFormat = true;
+
+    final compressedFile =
+        await FlutterImageCompress.compressWithFile(file.path);
+
+    if (compressedFile != null) {
+      final imageProvider = MemoryImage(compressedFile);
+
+      unawaited(
```

Comment:
We could reuse `_onFileLoaded` here?

Reply:
Got it, we could maybe extract the following bit into a single function?

```dart
await _updatePhotoViewController(
  previewImageProvider: _imageProvider,
  finalImageProvider: imageProvider,
);
setState(() {
  _imageProvider = imageProvider;
  _loadedFinalImage = true;
  _logger.info("Final image loaded");
});
```

Reply:
Done